### PR TITLE
If eslint not listed as dependency use global one only

### LINKED ISSUE #549: Don't use local eslint unless it's explicitly listed in package.json
If I leave `eslint.nodePath` set to `null` (the default), and I edit an NPM project that does not have `eslint` explicitly listed in its `package.json`, yet one of its deps includes `eslint`, then this VSC extension will still use the local eslint rather than the global one. This is not preferred, because then global plugins will not be loaded.

### PR DESCRIPTION
**Pull request by janowsiany**
Fixes #549

### CODE REVIEW

**Comment by janowsiany** [comment]
ping @dbaeumer

File: server/src/eslintServer.ts [hunk 1]
```diff
@@ -413,7 +413,7 @@ function resolveSettings(document: TextDocument): Thenable<TextDocumentSettings>
 					return Files.resolve('eslint', settings.resolvedGlobalPackageManagerPath, directory, trace);
 				});
 			} else {
-				promise = Files.resolve('eslint', settings.resolvedGlobalPackageManagerPath, directory, trace);
+				promise = Files.resolve('eslint', settings.resolvedGlobalPackageManagerPath, undefined, trace);
```
Comment (dbaeumer) [inline server/src/eslintServer.ts]: This only results in the fact the local installed eslint version is never considered even if it is not listed in the package.json.

**Comment by dbaeumer** [comment]
See also https://github.com/Microsoft/vscode-eslint/issues/549#issuecomment-440717379

**Comment by dbaeumer** [comment]
I will close the PR since it has no traction right now. Please ping if you want to continue on this.

**Pull request closed without merging**